californium-core/src/main/java/org/eclipse/californium/core/observe/Observation.java

```diff
@@ -52,6 +52,20 @@ public Request getRequest() {
 	}
 
 	/**
+	 * @return a cloned observation with a shallow clone of request
+	 */
+	public Observation cloneObservation() {
```

Comment:
Because of the **very specfic** "clone" of the Observation. A lot of information is not deep cloned and it was just done to get rid of the serialization/deserialization.
Mainly the InMemoryRegistrationStore in the leshan branch "improve_builders" remind me, that the current implementation seems to be fragile.